File: 12250 Language Detection/solution.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int cases = 0;
        String S;
        StringBuilder output = new StringBuilder();
        while((S = br.readLine()) != null && !S.equals("#")) {
            output.append("Case ").append(++cases).append(": ");
            if(S.equals("HELLO"))
                output.append("ENGLISH\n");
            else if(S.equals("HOLA"))
                output.append("SPANISH\n");
            else if(S.equals("HALLO"))
                output.append("GERMAN\n");
            else if(S.equals("BONJOUR"))
                output.append("FRENCH\n");
            else if(S.equals("CIAO"))
                output.append("ITALIAN\n");
            else if(S.equals("ZDRAVSTVUJTE"))
                output.append("RUSSIAN\n");
            else
                output.append("UNKNOWN\n");
        }
        System.out.print(output);
    }
}

File: 12250 Language Detection/solution.c
#include <stdio.h>
#include <string.h>

int main() {
    int cases = 0;
	char S[15];
	while(scanf("%s", S) && S[0] != '#') {
	    if(strcmp(S, "HELLO") == 0)
	        printf("Case %d: ENGLISH\n", ++cases);
	    else if(strcmp(S, "HOLA") == 0)
	        printf("Case %d: SPANISH\n", ++cases);
	    else if(strcmp(S, "HALLO") == 0)
	        printf("Case %d: GERMAN\n", ++cases);
	    else if(strcmp(S, "BONJOUR") == 0)
	        printf("Case %d: FRENCH\n", ++cases);
	    else if(strcmp(S, "CIAO") == 0)
	        printf("Case %d: ITALIAN\n", ++cases);
	    else if(strcmp(S, "ZDRAVSTVUJTE") == 0)
	        printf("Case %d: RUSSIAN\n", ++cases);
	    else
	        printf("Case %d: UNKNOWN\n", ++cases);
	}
	return 0;
}
